Tribler/Test/test_gui_dialogs.py

```diff
@@ -176,18 +173,27 @@ def disable_dispersy():
         self.startTest(disable_dispersy)
 
     def test_confirmationdialog(self):
-        def do_assert():
+
+        def cancel_dialog():
             dialog = wx.FindWindowByName('MFdialog')
-            self.assert_(isinstance(dialog, ConfirmationDialog), 'could not find ConfirmationDialog')
+            dialog.EndModal(wx.ID_CANCEL)
 
+        def screenshot():
+            dialog = wx.FindWindowByName('MFdialog')
             self.screenshot('Screenshot of ConfirmationDialog', window=dialog)
-            self.callLater(1, lambda: dialog.EndModal(wx.ID_CANCEL))
-            self.callLater(2, self.quit)
+
+        def do_assert():
+            dialog = wx.FindWindowByName('MFdialog')
+            self.assert_(isinstance(dialog, ConfirmationDialog), 'could not find ConfirmationDialog')
+            self.assert_(dialog.IsModal(), "ConfirmationDialog was not modal")
 
         def do_mark(item):
             self.assert_(isinstance(item, ChannelListItem), 'do_mark called without a ChannelListItem')
 
             self.callLater(10, do_assert)
+            self.callLater(11, screenshot)
+            self.callLater(13, cancel_dialog)
+            self.callLater(15, self.quit)
             self.guiUtility.MarkAsFavorite(None, item.original_data)
```

Comment:
@whirm Most likely not, but I will test this tomorrow. I first added this during debugging because I wanted the flow altered as I suspect the callLater still would trigger even if there was a showModal (which was added during debugging but later removed).